src/components/Sidebar/SidebarContainer.part.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import styled from '../../utils/styled';
+import { PositionType } from './SidebarPopup.types.part';
+
+export interface SidebarContainer {
+  open: boolean;
+  position?: PositionType;
+  size?: string;
+}
+
+export const SidebarContainer = styled.div<SidebarContainer>(
+  ({ open, position = 'right' as PositionType, size = '275px' }) => {
+    return `
+    transition: all 0.3s cubic-bezier(0, 0, 0.25, 1);
+    user-select: none;
+    position: fixed;
+    background-color: white;
+    z-index: 100001;
+    ${
+      !open
+        ? `
+        visibility: hidden;
+        opacity: 0;
+        overflow: initial;
+        `
+        : `
+        visibility: visible;
+        opacity: 1;
+        overflow: auto;
+        `
+    }
+    ${getSizeStyles(position, size)}
+    ${getPositonStyles(position, open, size)}
+  `;
+  },
+);
+
+function getSizeStyles(position: PositionType, size: string) {
+  if (position === 'top' || position === 'bottom') {
+    return `
+      height: ${size}; 
+      width: 100%;
+    `;
+  } else {
+    return `
+      width: ${size}; 
+      height: 100%;
+    `;
+  }
+}
+
+function getPositonStyles(position: PositionType, open: boolean, size: string) {
```

Comment:
did you mean `getPositionStyles`? 😊

Reply:
SidebarPopup has this functionality and can be accessible by Sidebar.inner.SidebarPopup